src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function add(num) {
+    if (num === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += num;
+
+      return add;
+    }
```

Comment:
```suggestion
    if (num === undefined) {
      const result = sum;

      sum = 0;

      return result;
    } 
    
    sum += num;

    return add;
    }
```
redundant else block